tensorflow/go/tensor.go

```diff
@@ -92,7 +92,7 @@ func NewTensor(value interface{}) (*Tensor, error) {
 	raw := tensorData(t.c)
 	buf := bytes.NewBuffer(raw[:0:len(raw)])
 	if dataType != String {
-		if err := encodeTensor(buf, val); err != nil {
+		if err := encodeTensor(buf, val, shape); err != nil {
```

Comment:
This means that we no longer do the check on string types, right?

Reply:
You are absolutely right, I missed this point.
Here is a fix for strings as well